mdata/aggmetric.go

```diff
@@ -356,7 +356,16 @@ func (a *AggMetric) persist(pos int) {
 	// create an array of chunks that need to be sent to the writeQueue.
 	pending := make([]*ChunkWriteRequest, 1)
 	// add the current chunk to the list of chunks to send to the writeQueue
-	cwr := NewChunkWriteRequest(a, a.key, chunk, a.ttl, a.chunkSpan, time.Now())
+	cwr := NewChunkWriteRequest(func() {
```

Comment:
Might be better to move the callback to just be `a.SyncChunkSaveState(chunk.Series.T0, true)`
and update SyncChunkSaveState to be a generator. eg.
```

// Generate the callback function for syncing chunk save state.
// 'ts' is the chunk T0 of the chunk that was saved, if sendPersist is true a 
// SendPersistMessage will be generated.
func (a *AggMetric) SyncChunkSaveState(ts uint32, sendPersist bool) ChunkSaveCallback {
	return func() {
		a.Lock()

		if ts > a.lastSaveFinish {
			a.lastSaveFinish = ts
		}
		if ts > a.lastSaveStart {
			a.lastSaveStart = ts
		}
		a.Unlock()
		log.Debugf("AM: metric %s at chunk T0=%d has been saved.", a.key, ts)
		if sendPersist {
			SendPersistMessage(a.key.String(), ts)
		}
	}
}
```